Makefile

```diff
@@ -169,10 +169,14 @@ cfn/lint: | guard/program/cfn-lint
 
 ## Runs eclint against the project
 eclint/lint: | guard/program/eclint guard/program/git
-eclint/lint: ECLINT_PREFIX ?= git ls-files -z | xargs -0
+eclint/lint: HAS_UNTRACKED_CHANGES ?= $(shell git diff-index --quiet HEAD -- || echo "true")
+eclint/lint: ECLINT_PREFIX ?= git ls-files -z
 eclint/lint:
 	@ echo "[$@]: Running eclint..."
-	cd $(PROJECT_ROOT) && $(ECLINT_PREFIX) eclint check
+	cd $(PROJECT_ROOT) && \
+	[ -z "$(HAS_UNTRACKED_CHANGES)" ] && \
+	($(ECLINT_PREFIX) | $(XARGS) --null bash -c 'if ! [[ "{}" == *".bats"* ]]; then eclint check {}; fi') || \
```

Comment:
Would it be easier to filter the `git ls-files -z` command? e.g. `git ls-files -z | grep -v .bats`

```
$(ECLINT_FILES) | $(XARGS) --null eclint check {}
```

Reply:
went down that route at first but couldn't get this to work properly: 
`git ls-files | grep -v ".bats" | eclint check` 
eclint still manages to process `.bats` files

Reply:
`eclint check` != `eclint check {}`... that just an incomplete example, or is it... everything?

Reply:
tracking, was actually using `git ls-files -z | grep -v ".bats" | xargs -0 eclint check` but `git ls-files | grep -v ".bats" | xargs -I {} eclint check {}` looks like its working as expected. will update here shortly

Reply:
If you want to keep the nul support, grep -z works:

```
git ls-files -z | grep -zv ".bats" | xargs -0 -I {} eclint check {}
```

Reply:
had to use gnu grep to get this to work.. thought I fixed most of the default bash tools on macos but apparently not. was previously running `grep (BSD grep) 2.5.1-FreeBSD` switched to `grep (GNU grep) 3.4` and things work as expected